library/core/src/ptr/non_null.rs

```diff
@@ -124,7 +124,7 @@ impl<T: Sized> NonNull<T> {
     #[must_use]
     #[unstable(feature = "ptr_as_uninit", issue = "75402")]
     #[rustc_const_unstable(feature = "const_ptr_as_ref", issue = "91822")]
-    pub const unsafe fn as_uninit_ref<'a>(&self) -> &'a MaybeUninit<T> {
+    pub const unsafe fn as_uninit_ref<'a>(self) -> &'a MaybeUninit<T> {
```

Comment:
`NonNull::as_ref` takes `&Self`.  It would seem odd for `NonNull::as_uninit_ref` to diverge from that?

Reply:
Ah sorry this is already being discussed here, never mind.

Reply:
To be clear, I don't feel that strongly and the consistency argument may be more important.